editoast/src/schema/timetable.rs

```diff
@@ -0,0 +1,62 @@
+use super::{TrainSchedule, TrainScheduleDetails};
+use crate::error::{InternalError, Result};
+use crate::tables::osrd_infra_timetable;
+use diesel::pg::PgConnection;
+use diesel::prelude::*;
+use diesel::result::Error as DieselError;
+use editoast_derive::EditoastError;
+use serde::Serialize;
+use thiserror::Error;
+
+#[derive(Debug, PartialEq, Queryable, Identifiable, Serialize, Selectable)]
+#[diesel(table_name = osrd_infra_timetable)]
+pub struct Timetable {
+    pub id: i64,
+    pub name: String,
+}
+
+#[derive(Debug, Error, EditoastError)]
+#[editoast_error(base_id = "timetable")]
+pub enum TimetableError {
+    #[error("Timetable '{timetable_id}', could not be found")]
+    #[editoast_error(status = 404)]
+    NotFound { timetable_id: i64 },
+}
+
+#[derive(Debug, PartialEq, Serialize)]
+pub struct TimetableWithSchedules {
+    pub id: i64,
+    pub name: String,
+    pub train_schedules: Vec<TrainScheduleDetails>,
+}
+
+impl Timetable {
+    /// Retrieves timetable with a specific id and its associated schedules
+    pub fn retrieve(conn: &mut PgConnection, timetable_id: i64) -> Result<TimetableWithSchedules> {
+        let timetable = match osrd_infra_timetable::table
+            .filter(osrd_infra_timetable::id.eq(timetable_id))
+            .select(Timetable::as_select())
+            .get_result(conn)
+        {
+            Ok(timetable) => Ok::<Timetable, InternalError>(timetable),
+            Err(DieselError::NotFound) => Err(TimetableError::NotFound { timetable_id }.into()),
+            Err(e) => Err(e.into()),
+        }?;
+
+        let train_schedules = match TrainSchedule::belonging_to(&timetable)
+            .select(TrainSchedule::as_select())
+            .load::<TrainSchedule>(conn)
+        {
+            Ok(train_schedules) => Ok::<Vec<TrainSchedule>, InternalError>(train_schedules),
+            Err(e) => Err(e.into()),
+        }?;
+        Ok(TimetableWithSchedules {
+            id: timetable.id,
+            name: timetable.name,
+            train_schedules: train_schedules
+                .into_iter()
+                .map(|schedule| schedule.as_details())
+                .collect(),
+        })
+    }
+}
```

Comment:
You should add a test to this function

Reply:
I think I need a scenario to create a timetable, I have added an integration test comparing `api` and `editoast` results as we are waiting for your work on scenario. @flomonster is it ok for you if we only have an integration test for now